src/arkode/arkode_io.c

```diff
@@ -918,6 +918,79 @@ int ARKodeSetAdaptController(void* arkode_mem, SUNAdaptController C)
   return (arkReplaceAdaptController(ark_mem, C, SUNFALSE));
 }
 
+/*---------------------------------------------------------------
+  ARKodeSetAdaptControllerByName:
+
+  Specifies a SUNAdaptController time step controller object by
+  its name.
+  ---------------------------------------------------------------*/
+int ARKodeSetAdaptControllerByName(void* arkode_mem, const char* cname)
+{
+  int retval;
+  ARKodeMem ark_mem;
+  if (arkode_mem == NULL)
+  {
+    arkProcessError(NULL, ARK_MEM_NULL, __LINE__, __func__, __FILE__,
+                    MSG_ARK_NO_MEM);
+    return (ARK_MEM_NULL);
+  }
+  ark_mem = (ARKodeMem)arkode_mem;
+
+  /* Create new controller based on the name */
+  SUNAdaptController C = NULL;
+  if (strcmp(cname, "Soderlind") == 0)
+  {
+    C = SUNAdaptController_Soderlind(ark_mem->sunctx);
+  }
+  else if (strcmp(cname, "PID") == 0)
+  {
+    C = SUNAdaptController_PID(ark_mem->sunctx);
+  }
+  else if (strcmp(cname, "PI") == 0)
+  {
+    C = SUNAdaptController_PI(ark_mem->sunctx);
+  }
+  else if (strcmp(cname, "I") == 0)
+  {
+    C = SUNAdaptController_I(ark_mem->sunctx);
+  }
+  else if (strcmp(cname, "ExpGus") == 0)
+  {
+    C = SUNAdaptController_ExpGus(ark_mem->sunctx);
+  }
+  else if (strcmp(cname, "ImpGus") == 0)
+  {
+    C = SUNAdaptController_ImpGus(ark_mem->sunctx);
+  }
+  else if (strcmp(cname, "H0211") == 0)
+  {
+    C = SUNAdaptController_H0211(ark_mem->sunctx);
+  }
+  else if (strcmp(cname, "H211") == 0)
+  {
+    C = SUNAdaptController_H211(ark_mem->sunctx);
+  }
+  else if (strcmp(cname, "H312") == 0)
+  {
+    C = SUNAdaptController_H312(ark_mem->sunctx);
+  }
```

Comment:
Should this be H0312 or H312?
```suggestion
  else if (strcmp(cname, "H0312") == 0)
  {
    C = SUNAdaptController_H0312(ark_mem->sunctx);
  }
```